### PR TITLE
Allow sending additional trailers in error response

### PR DESCRIPTION
**Pull request by janchochol**

### CODE REVIEW

**Comment by janchochol** [comment]
After #31, we found that we also need to add custom trailers to error response (it is possible to do it for ok response through `Ctx`, but there was no way to do it for errors).
This is probably easiest way to do it, and due to planned refactor of client, I do not think, that it needs more investment.

**Comment by janchochol** [comment]
Hi @tsloughter, any progress reviewing this PR?

**Comment by tsloughter** [comment]
Sorry, I missed this PR. Is the issue that in a unary call you don't have a stream so can't use `grpcbox_stream:add_trailers/2` to add trailers before returning an error?

**Comment by janchochol** [comment]
No problem.
Yes, something like `grpcbox_stream:add_trailers/2` in unary calls will also help.

**Comment by janchochol** [comment]
Hi @tsloughter, any progress?
We are using custom branch, but we would like to switch to upstream at some time.

**Comment by tsloughter** [comment]
Hey, yea, sorry, I'll get this merged today and make a new release.

File: src/grpcbox_stream.erl [hunk 1]
```diff
@@ -209,6 +217,10 @@ on_receive_data(Bin, State=#state{request_encoding=Encoding,
     catch
         throw:{grpc_error, {Status, Message}} ->
             end_stream(Status, Message, State);
+        throw:{grpc_extended_error, #{status := Status, message := Message} = ErrorData} ->
+            Trailers = maps:get(trailers, ErrorData, #{}),
+            State2 = update_trailers(maps:to_list(Trailers), State),
+            end_stream(Status, Message, State2);
```
Comment (tsloughter) [inline src/grpcbox_stream.erl:223]: Make into a separate function to be used in `handle_info('EXIT'...` as where the `maps:get` had the map and key transposed.

**Comment by tsloughter** [comment]
@janchochol ping, did you see my comments about some needed changes before I can merge?

**Comment by janchochol** [comment]
Sorry for not responding - I was little bit too busy at work. I saw them, I will add missing test.

**Comment by janchochol** [comment]
Test added and function added.
I also split it 2 commits, as I am not sure if it is worth to add `grpc_extended_error` to output streaming (this is using `handle_info('EXIT'...` code), as it can be achieved using existing functionality.

**Comment by tsloughter** [comment]
Thanks! This is good.

Going to merge this and cut a release today. 

I think I get what you mean in your last comment but I'm not positive. The trailers were possible in the streaming output already but you added the `EXIT` handling `grpc_extended_error` in your original commit, right?

**Pull request merged**